crates/generator/src/syscalls/mod.rs

```diff
@@ -524,6 +529,76 @@ impl<'a, 'b, S: State, C: ChainView, Mac: SupportMachine> Syscalls<Mac>
                 machine.set_register(A0, Mac::REG::from_u8(SUCCESS));
                 Ok(true)
             }
+            SYS_BN_ADD => {
+                let input_addr = machine.registers()[A3].to_u64();
+                let input_size = machine.registers()[A4].to_u64();
+                let input = load_bytes(machine, input_addr, input_size as usize)?;
+
+                let output = bn::add(&input).map_err(|err| {
+                    let err_msg = format!("syscall SYS_BN_ADD error: {:?}", err.0);
+                    log::error!("{}", err_msg);
+                    VMError::Unexpected(err_msg)
+                })?;
+                store_data(machine, output.as_slice())?;
+
+                machine.set_register(A0, Mac::REG::from_u8(SUCCESS));
+                Ok(true)
+            }
+            SYS_BN_MUL => {
+                let input_addr = machine.registers()[A3].to_u64();
+                let input_size = machine.registers()[A4].to_u64();
+                let input = load_bytes(machine, input_addr, input_size as usize)?;
+
+                let output = bn::mul(&input).map_err(|err| {
+                    let err_msg = format!("syscall SYS_BN_MUL error: {:?}", err.0);
+                    log::error!("{}", err_msg);
+                    VMError::Unexpected(err_msg)
+                })?;
+                store_data(machine, output.as_slice())?;
+
+                machine.set_register(A0, Mac::REG::from_u8(SUCCESS));
+                Ok(true)
+            }
+            SYS_BN_PAIRING => {
+                let input_addr = machine.registers()[A3].to_u64();
+                let input_size = machine.registers()[A4].to_u64();
+                let input = load_bytes(machine, input_addr, input_size as usize)?;
+
+                if let Some(cycles_pool) = self.cycles_pool {
+                    // k is the number of pairings being computed
+                    let k: u64 = input_size / 192;
+                    if k > 0 {
+                        // Subtract additional cycles per pairing
+                        // https://eips.ethereum.org/EIPS/eip-1108
+                        let additional_cycles = k * 34000 * 3;
```

Comment:
Can we add this modifier factor to `SyscallCyclesConfig`?

Reply:
https://github.com/godwokenrises/godwoken/pull/808/commits/61b58b1d232570f054345c55c7fcf821338607c8